bundler/lib/bundler/cli/init.rb

```diff
@@ -27,25 +28,43 @@ def run
 
         spec = Bundler.load_gemspec_uncached(gemspec)
 
-        File.open(gemfile, "wb") do |file|
+        File.open(@gemfile, "wb") do |file|
           file << "# Generated from #{gemspec}\n"
           file << spec.to_gemfile
         end
       else
-        File.open(File.expand_path("../templates/#{gemfile}", __dir__), "r") do |template|
-          File.open(gemfile, "wb") do |destination|
+        File.open(File.expand_path(template_path, __dir__), "r") do |template|
+          File.open(@gemfile, "wb") do |destination|
             IO.copy_stream(template, destination)
           end
         end
       end
 
-      puts "Writing new #{gemfile} to #{SharedHelpers.pwd}/#{gemfile}"
+      puts "Writing new #{@gemfile} to #{SharedHelpers.pwd}/#{@gemfile}"
     end
 
     private
 
-    def gemfile
-      @gemfile ||= Bundler.preferred_gemfile_name
+    def preferred_gemfile_name
+      gemfile_name = options[:gemfile] || Bundler.preferred_gemfile_name
+
+      if File.exist?(gemfile_name) && gemfile_name != "Gemfile"
+        Bundler.ui.warn "Bundler will name the gemfile \"Gemfile\" because " \
+                        "#{gemfile_name} already exists at #{File.expand_path(gemfile_name)}. " \
+                        "To silence this warning, run `bundle init --gemfile Gemfile`."
+        "Gemfile"
+      else
+        gemfile_name
+      end
+    end
+
+    def template_path
+      template_folder = "../templates/"
```

Comment:
```suggestion
      template_file = "../templates/#{@gemfile}"
      return template_file if File.exist?(template_file)
      
      "../templates/Gemfile"
```

Reply:
Thanks for the review!

Reply:
Was this a bug, or something else? If a bug that wasn't caught by CI, would be nice to add an extra spec.

Reply:
No, it's not a bug. It's just a simpler way to set the template file.

Reply:
Thanks for clarifying!